Orange/canvas/document/quickmenu.py

```diff
@@ -323,7 +323,20 @@ def filterFunc(self):
         return self.__filterFunc
 
     def filterAcceptsRow(self, row, parent=QModelIndex()):
-        accepted = QSortFilterProxyModel.filterAcceptsRow(self, row, parent)
+        flat_model = self.sourceModel()
+        index = flat_model.index(row, self.filterKeyColumn(), parent)
+        description = flat_model.data(index, role=QtWidgetRegistry.WIDGET_DESC_ROLE)
+        name = description.name
+        keywords = description.keywords
+
+        # match name and keywords
+        accepted = False
+        for keyword in [name] + keywords:
```

Comment:
How about also iterating over `filter(None, re.split(r'\W+', description.description.lower()))`? This way, many widgets won't even need separate keywords, e.g. K-means already has "clustering", "silhouette", ...

Reply:
I'd rather manually set keywords to be able to finely control what shows up in search, avoiding the overcrowding of search results. Each widget shouldn't need more than a few keywords I think.